pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -392,22 +395,27 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundListeners(env *model.E
 				protocol:       servicePort.Protocol,
 			}
 
-			currentListener = nil
+			currentListenerEntry = nil
 
 			switch plugin.ModelProtocolToListenerProtocol(servicePort.Protocol) {
 			case plugin.ListenerProtocolHTTP:
 				listenerMapKey = fmt.Sprintf("%s:%d", listenAddress, servicePort.Port)
-				if current, exists := listenerMap[listenerMapKey]; exists {
-					if !current.servicePort.Protocol.IsHTTP() {
-						env.PushStatus.Add(model.ProxyStatusConflictHTTPOut,
+				var exists bool
+				// Check if this HTTP listener conflicts with an existing wildcard TCP listener
+				// i.e. one of NONE resolution type, since we collapse all HTTP listeners into
+				// a single 0.0.0.0:port listener and use vhosts to distinguish individual http
+				// services in that port
+				if currentListenerEntry, exists = listenerMap[listenerMapKey]; exists {
+					if !currentListenerEntry.servicePort.Protocol.IsHTTP() {
+						env.PushStatus.Add(model.ProxyStatusConflictOutboundListenerTCPOverHTTP,
 							listenerMapKey, node,
-							fmt.Sprintf("Port=%d Accepted=%s Rejected=%s Key=%s",
-								current.servicePort.Port,
-								current.service.Hostname,
-								service.Hostname,
-								listenerMapKey))
+							fmt.Sprintf("Listener=%s HTTPServices=%d RejectedTCP=%s",
```

Comment:
hmm.. if its multiple services, its helpful to dump everything out in the log rather than just the first one. I don't mind concatenating all hostnames into one giant string, as long as this metric stuff will work.

Reply:
sgtm